src/function/scalar/list/list_reverse.cpp

```diff
@@ -0,0 +1,124 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_parameter_expression.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+
+namespace duckdb {
+
+static void ListReverseFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	auto count = args.size();
+
+	Vector &input_list = args.data[0];
+	if (input_list.GetType().id() == LogicalTypeId::SQLNULL) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		ConstantVector::SetNull(result, true);
+		return;
+	}
+
+	// get the child vector
+	auto input_list_list_size = ListVector::GetListSize(input_list);
+	auto &input_list_child = ListVector::GetEntry(input_list);
+
+	UnifiedVectorFormat input_list_data;
+	input_list.ToUnifiedFormat(count, input_list_data);
+	auto input_list_entries = (list_entry_t *)input_list_data.data;
+
+	// get the child data
+	UnifiedVectorFormat input_list_child_data;
+	input_list_child.ToUnifiedFormat(input_list_list_size, input_list_child_data);
+
+	result.SetVectorType(VectorType::FLAT_VECTOR);
+	auto result_entries = FlatVector::GetData<list_entry_t>(result);
+	auto &result_validity = FlatVector::Validity(result);
+
+	// create a selection vector for slicing the child vector
+	SelectionVector rev_sel(input_list_list_size);
+
+	idx_t offset = 0;
+	idx_t offset_per_new_row = 0;
+
+	for (idx_t i = 0; i < count; i++) {
+		auto input_list_list_index = input_list_data.sel->get_index(i);
+
+		if (!input_list_data.validity.RowIsValid(input_list_list_index)) {
+			result_validity.SetInvalid(i);
+			continue;
+		};
+
+		result_entries[i].offset = offset;
+		result_entries[i].length = 0;
+
+		D_ASSERT(input_list_data.validity.RowIsValid(input_list_list_index));
+		const auto &input_list_entry = input_list_entries[input_list_list_index];
+		result_entries[i].length += input_list_entry.length;
+
+		// set reverse selection vector indices
+		// set index of selection vector in a way, that only the entries of the current row are reversed at once
+		for (idx_t j = 0; j < input_list_entry.length; j++) {
+			rev_sel.set_index(j + offset_per_new_row, input_list_entry.length - j - 1 + offset_per_new_row);
```

Comment:
the `offset_per_new_row` is the reason you were having difficulties with VectorType's other than FLAT, because ConstantVectors only have 1 physical row, but `count` logical rows, this would overflow the selection vector.

If you instead use `input_list_entry.offset` for this, that issue is solved

Reply:
Thats it! That helps a lot. Thank you so much!!!